File: back_endl/app/Models/Offer.php
<?php

namespace App\Models;


use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Factories\HasFactory;

class Offer extends Model
{     use HasFactory;
    
    protected $fillable = [
        'user_id',
    'price',
    'bedrooms',
    'addres',
    'area',
    'bathrooms',
    'kitchen',
    'garage',
    'description',
    
];
    //
    public function user(){
        return $this->belongsTo(agence::class);
    }
   
  
    public function comments()
    {
        return $this->hasMany(Comment::class);
    }
    public function images()
    {
        return $this->hasMany(Image::class);
    }

}

File: front_endf/lib/models/Offer.dart
class Offer {
  int? id;
  String? numberphone;
  String? price;
  String? bedrooms;
  String? addres;
  String? area;
  String? bathrooms;
  String? kitchen;
  String? garage;
  String? description;
  String? operation;
  String? latitude;
  String? longitude;
  String? name_agence;
  String? categorie;
  String? image;

  Offer({
    this.id,
    this.numberphone,
    this.price,
    this.bedrooms,
    this.addres,
    this.area,
    this.bathrooms,
    this.kitchen,
    this.garage,
    this.description,
    this.image,
    this.operation,
    this.latitude,
    this.longitude,
    this.name_agence,
    this.categorie,
  });

  factory Offer.fromJson(Map<String, dynamic> json) {
    return Offer(
      id: json['id'],
      numberphone: json['numberphone'],
      price: json['price'],
      bedrooms: json['bedrooms'],
      addres: json['addres'],
      area: json['area'],
      bathrooms: json['bathrooms'],
      kitchen: json['kitchen'],
      garage: json['garage'],
      description: json['description'],
      operation: json['operation'],
      name_agence: json['name_agence'],
      categorie: json['categorie'],
      latitude: json['latitude'],
      longitude: json['longitude'],
    );
  }
}
